scripts/tableland/tasks.ts

```diff
@@ -497,10 +497,227 @@ task('create-manufacturer-table-schema', 'npx hardhat create-manufacturer-table-
+                    .createDeviceDefinitionTable(tableOwner, manufacturerId, { gasPrice: _gasPrice });
+
+                await tablelandValidator.pollForReceiptByTransactionHash({
+                    chainId: CHAIN_ID[currentNetwork],
+                    transactionHash: (await dTx.wait())?.hash as string,
+                });
+
+                const ddTableName = await ddTableInstance.getDeviceDefinitionTableName(manufacturerId);
+
+                console.log(`Device Definition table created\nTable ID: ${ddTableId}\nTable Name: ${ddTableName}`);
+            }
+
+            ddTableId = await ddTableInstance.getDeviceDefinitionTableId(manufacturerId);
+
+            const deviceDefinitionByManufacturers = devices.filter((c) => c.make.name === make && c.type.year > 2006);
+            const ddTableName = await ddTableInstance.getDeviceDefinitionTableName(manufacturerId);
+            const tablelandDeviceDefinitionByManufacturers = await getDeviceDefinitionsByTableName(db, ddTableName);
+
+            console.log(`API Device Definition By Manufacturer [${make}] total => ${deviceDefinitionByManufacturers.length}`);
+            console.log(`Tableland Device Definition By Manufacturer [${make}] total => ${tablelandDeviceDefinitionByManufacturers.length}`);
+
+            // Insert new dd
+            let newDeviceDefinitionByManufacturers = [];
+            deviceDefinitionByManufacturers.forEach(element => {
+                const dds = tablelandDeviceDefinitionByManufacturers.filter((c) => c.id === element.name_slug);
+                if (dds != undefined && dds.length == 0){
+                    newDeviceDefinitionByManufacturers.push(element);
+                }
+            });
+
+            // Update dd when change attributes
+            let updateDeviceDefinitionByManufacturers = [];
+            deviceDefinitionByManufacturers.forEach(element => {
+                const dds = tablelandDeviceDefinitionByManufacturers.filter((c) => c.id === element.name_slug);
+                if (dds != undefined && dds.length > 0){
+
+                    if (dds[0].metadata.device_attributes) {
+                        const different = dds[0].metadata.device_attributes.filter(obj2 => {
+                            const obj1 = element.device_attributes.find(item => item.name === obj2.name);
+                            return obj1 && obj1.value !== obj2.value;
+                        });
+                          
+                        if (different.length > 0) {
+                            updateDeviceDefinitionByManufacturers.push(element);
+                        }
+                    }
+                    
+                }
+            });
+
+            // delete dd
+            let deleteDeviceDefinitionByManufacturers = [];
+            deviceDefinitionByManufacturers.forEach(element => {
+                const dds = tablelandDeviceDefinitionByManufacturers.filter((c) => c.ksuid === element.device_definition_id);
```

Comment:
hmm actually I think this should be the other way around:
- iterate over tableland defs
- for each lookup in deviceDefinitions
- if not found, delete from tableland.

Reply:
I mean I guess this would work, and is accounting for differences in the id to delete. But there are certain scenarios where i think we wouldn't be deleting